### PR TITLE
VSS: ensure an out-of-band Secret deletion is properly remediated

### LINKED ISSUE #124: K8S secret is not re-created if deleted for some bad reason
**Describe the bug**
The K8S is not re-created during the sync phase of the vaultstaticsecret.

**To Reproduce**
1. create a vaultstatic secret
2. wait the K8S secret creation
3. drop the K8S secret generated
5. on the operator log we can see the sync log:
```
 DEBUG    events    Secret sync not required    {"type": "Normal", "object": {"kind":"VaultS │
│ taticSecret","namespace":"minio","name":"minio-encrypted-aws-root-key","uid":"98bb47f6-7aa5-41f3-b04c-45c1ae2d5ebe","apiVer ││ sion":"secrets.hashicorp.com/v1alpha1","resourceVersion":"196528971"}, "reason": "SecretSync"}
```

But the K8S secret is not more existing

**Expected behavior**
the K8S secret is re-created

**Environment**
* Kubernetes version:  1.22.2
* vault-secrets-operator version: 0.11.0

### PR DESCRIPTION
**Pull request by benashz**
The Operator now detects whenever the owned Destination Secret is deleted. Whenever this situation is encountered, the Operator will recreate/repopulate the Secret.

RefreshAfter must be set and Destination.Create must be set to true for the remediation to take place.

Closes #124

### CODE REVIEW

File: test/integration/vaultstaticsecret_integration_test.go [hunk 1]
```diff
@@ -367,6 +366,19 @@ func TestVaultStaticSecret_kv(t *testing.T) {
 			} else {
 				assertNoHMAC(t, obj)
 			}
+
+			if obj.Spec.Destination.Create {
+				sec, _, err := helpers.GetSecret(ctx, crdClient, obj)
+				if assert.NoError(t, err) {
+					// ensure that a Secret deleted ouf of band is properly restored
```
Comment (adrianmoisey) [inline test/integration/vaultstaticsecret_integration_test.go:373]: Nitpick: 
```suggestion
					// ensure that a Secret deleted out of band is properly restored
```

Comment (benashz) [reply test/integration/vaultstaticsecret_integration_test.go]: Fixed in https://github.com/hashicorp/vault-secrets-operator/pull/137/commits/e3bda36c3b3443478109248c200fe0926896a9a5

**Review by tvoran — APPROVED** [review]
Seems to work for me!

File: controllers/vaultstaticsecret_controller.go [hunk 2]
```diff
@@ -230,6 +230,8 @@ func (r *VaultStaticSecretReconciler) handleSecretHMAC(ctx context.Context, o *s
 			}
 
 			macsEqual = valid
+		} else {
+			macsEqual = false
```
[the remark this answers is not in the document]
Comment (benashz) [reply controllers/vaultstaticsecret_controller.go:235]: Added in https://github.com/hashicorp/vault-secrets-operator/pull/137/commits/82e206acee71b1208847d7b061ef44989e396055

**Pull request merged**